homeassistant/components/anova/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The Anova integration."""
+from __future__ import annotations
+
+from anova_wifi import AnovaPrecisionCooker, AnovaPrecisionCookerSensor
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN
+from .coordinator import AnovaCoordinator
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Anova from a config entry."""
+    device = AnovaPrecisionCooker(
```

Comment:
If there is an endpoint to poll per device, you can create multiple coordinators and store them in a dict indexed by the device id

https://github.com/home-assistant/core/pull/86254/files#diff-f9e6d6657840eb8ce3240c05de3201db80039febf298cdec9c0b5ebe2685e190R42

Reply:
Done - I'm not sure if I did it correctly though, as I don't get what the key part of the dictionary is used for, as the coordinator has the device in it

Reply:
Could I just swap it to a list? Or is there a benefit to a dictionary/ am I doing something that isn't ideal and that is why I have no use for the key

Reply:
Usually we put everything into a dataclass so when more is added later its easy to maintain

`lookin/models` has an example

Reply:
Done - Let me know if you think how I implemented it is fine, I added some data that isn't explicitly and I labeled the devices as "precision_cookers" instead of just devices, as another user wants to develop support for the oven in the future, so by having a list between precision_cookers and ovens, it will make things easier when we need to handle them differently.